docs/src/stories/components/Layout/LayoutBeta.stories.jsx

```diff
@@ -0,0 +1,485 @@
+import React from 'react'
+import clsx from 'clsx'
+import {NavWithSubItems} from '../ActionList/ActionListPatterns.stories'
+import PageLayoutBehavior from '../../helpers/pageLayoutBehavior.jsx'
+
+export default {
+  title: 'Components/Layout/Beta',
+  excludeStories: ['LayoutTemplate'],
+  argTypes: {
+
+    // Debug
+
+    _debug: {
+      control: 'boolean',
+      description: 'Show background colors in regions for debugging',
+    },
+
+    // Structure
+
+    wrapperSizing: {
+      options: ['fluid', 'md', 'lg', 'xl'],
+      control: {
+        type: 'inline-radio',
+        labels: ['fluid', 'md', 'lg', 'xl']
+      },
+      description: 'Define the maximum width of the component. `fluid` sets it to full-width. Other values center `Layout` horizontally. Refer to [container utilities](https://primer.style/css/objects/grid#containers) for reference.',
+      table: {
+        category: 'Structure'
+      }
+    },
+    outerSpacing: {
+      options: ['none', 'normal', 'condensed'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'Sets wrapper margins surrounding the component to distance itself from the viewport edges. `normal` sets the margin to 16px, and to 24px on `lg` breakpoints and above. `condensed` keeps the margin at 16px. `none` sets the margin to 0.',
+      table: {
+        category: 'Structure'
+      }
+    },
+    innerSpacing: {
+      options: ['none', 'normal', 'condensed'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'Sets padding to regions individually. `normal` sets padding to 16px, with the `content` region getting 24px horizontal padding on `lg` breakpoints and above. `condensed` keeps the padding always at `16px`. `none` sets the padding to 0.',
+      table: {
+        category: 'Structure'
+      }
+    },
+    columnGap: {
+      options: ['none', 'normal', 'condensed'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'Sets the gap between columns to distance them from each other. `normal` sets the gap to 16px, and to 24px on `lg` breakpoints and above. `condensed` keeps the gap always at 16px. `none` sets the gap to 0.',
+      table: {
+        category: 'Structure'
+      }
+    },
+    rowGap: {
+      options: ['none', 'normal', 'condensed'],
+      control: {
+        type: 'inline-radio'
+      },
+      description: 'Sets the gap below the header and above the footer. `normal` sets the gap to 16px, and to 24px on `lg` breakpoints and above. `condensed` keeps the gap always at 16px. `none` sets the gap to 0.',
+      table: {
+        category: 'Structure'
+      }
+    },
+
+    // Responsive
+
+    responsiveVariant: {
```

Comment:
I think we could improve the naming for things referring to the "responsive" styles. "Responsive" is vague and confusing because the whole component is responsive, but I think here we're only using it to refer to a specific breakpoint.

Any ideas on something that's more descriptive? If we only want "responsiveVariant" to get activated at a certain breakpoint, maybe we could just name it after the breakpoint?

For example:
| current                  | new                   |
| ------------------------ | --------------------- |
| `responsiveVariant`      | `variantAtSmall`      |
| `paneResponsivePosition` | `panePositionAtSmall` |
| `paneResponsiveDivider`  | `paneDividerAtSmall`  |
| etc...                   | etc...                |

Reply:
(related to https://github.com/github/primer/pull/463#pullrequestreview-810656271 :))

@mperrotti I agree `responsiveVariant` and is vague. (did I say naming is hard already today? :)), but looking at our options I still don't think it makes sense to include the breakpoint size in the prop name.

Existing components in Primer CSS use breakpoint names to indicate styles that are applied to their _breakpoint an up_ ([example with display](https://primer.style/css/utilities/layout#responsive-display)).

In the case of `PageLayout`, the responsive variant is applied up to `md`. `variantAtMedium` or `variantAtMd` sounds quite confusing, and even if we went that route,`variantAtMd` wouldn't be correct because it'd behave as "up to `md`", while the name would indicate "only at `md`" or "`md` and up".

Maybe we could have an actual naming for describing what `PageLayout`'s """mobile"""-friendly variant is. Something that doesn't conflict with the breakpoint names and indicates the viewport only accepts a single column... maybe... `narrowVariant`??

A quick search shows me "narrow screens" is a common terminology when talking about responsive design.


current | new
-- | --
`responsiveVariant` | `narrowVariant`
`paneResponsivePosition` | `panePositionAtNarrow`
`paneResponsiveDivider` | `paneDividerAtNarrow`
etc... | etc...

Reply:
Since Layout is responsive by default, I'm going to assume `stackRegions` is the default option for narrow layout behavior. With that being a `default` I wonder if we can combine the responsive specific props with the default props, for example:

prop | description | options
-- | -- | --
`panePosition` | Defines the position of the pane. start renders the pane on the left, and end renders it on the right. | `start` `end`
`paneResponsivePosition` | If responsiveVariant is set to stackRegions, defines the position of the pane in the responsive variant. start puts the pane above content, and end puts it below content. inherit uses the same value from panePosition. | `inherit` `start` `end`

Those two props essentially have the same options, and the difference depends on the stackRegions option from another prop. Its not clear to me why we need to give more granular options here, and means the user has to define more props. If we removed the responsive specific props, the results of the default props `panePosition` could more simply respond to the behavioral prop by default (which can be documented). 

Example:

prop | description | options
-- | -- | --
`panePosition` | Defines the position of the pane. For wider screens, `start` renders the pane on the left and `end` on the right. For narrow screens with stacked regions, `start` puts the pane above `content` and `end` puts it below `content` | `start` `end`

This style would eliminate `paneResponsivePosition` `paneResponsiveDivider` `headerResponsiveDivider` `footerResponsiveDivider`

Okay... so if we start to combine "responsive" logic with default props, what do we do with `responsiveVariant` and `responsivePrimaryRegion`?

Maybe just think of it as one variant where default is `stacked`
prop | description | options
-- | -- | --
`separateRegions` | defines how the layout component adapts to smaller viewports. By default Layout stacks region content in a vertical flow for small viewports, with pane and content vertically arranged. `separateRegions` presents pane and content as different pages on smaller viewports. Change the preview size from the toolbar to test it.  | true/false
`primaryRegion` | required if `separateRegions` is `true` | `content` `pane`

I'm thinking about this from a React perspective. I understand why the props are very granular right now, that's how we build things! But it can lead to confusion and user error if we expect the end user to apply a lot of props. So just thinking we might be able to simplify, make some assumptions with more defaults and only offer granularity when we really need it (like to set the primary region).

I hope this is helpful and not MORE confusing!

Reply:
> Since Layout is responsive by default, I'm going to assume `stackRegions` is the default option for narrow layout behavior.

That's correct 😎 !

#### _On `panePosition` and `paneResponsivePosition`:_

> Those two props essentially have the same options, and the difference depends on the `stackRegions` option from another prop. Its not clear to me why we need to give more granular options here, and means the user has to define more props.

By default `paneResponsivePosition` (or `panePositionAtNarrow` like is was suggested above) is always `inherit`, so it's not something people will have to worry when they're using the component. The reason I'm considering this extra _responsive_ property is for when the ordering on larger viewports doesn't match smaller ones.

On larger viewports, navigation/filtering sidebars should be on the left, while metadata/details should appear on the right. On mobile/narrow viewports, I don't think we should assume that it would always translate to "nav/filter at the top" and "metadata/details" at the bottom.

Per instance, due to the length the `content` region can reach in an Issues detail page, it might make more sense to include metadata details at the top (especially once we consolidate these sidebars using `ActionList`-based specialized components that might collapse extra items). Example with an old mockup:

> ![Mockup showing an issue detail page on mobile where the pane region appears above the content region.](https://user-images.githubusercontent.com/293280/146827057-eb425a03-21ea-4dec-a89a-3d0e6d551763.png)

I recognize this is an accessibility problem if we don't get the DOM ordering aligned. We could simplify it for now and remove it, but I believe this layout granularity will be needed at some point. c/c @ichelsea 

#### On responsive dividers

> This style would eliminate `paneResponsivePosition` `paneResponsiveDivider` `headerResponsiveDivider` `footerResponsiveDivider`

So, on narrow viewports the divider have an extra functionality, that's why these extra options exist 😇. While on larger screens the regions have a true/false option for showing/hiding dividers, on narrow viewports they can be rendered either as `line` or `filled`. This extra granularity helps complex pages provide a thicker separator in regions, like in the mockup above.

I agree the responsive dividers should have `inherit` options by default! I'll update it in code! 🙇 

(The question on how to call these _narrow viewport_-specific props remains) 😅

#### On `responsiveVariant` / boolean `separateRegions`

I like the simplicity of a boolean `spearateRegions`, but I do wonder what terminology we should be using for referring to these customizable "responsive variants" in general 🤔. Per instance, if Layout supported 3 different modes on narrow viewports, how should we call it? That's a problem we might face with Overlays (that should be able to render as bottom sheets or full-screen on narrow screens). I think this responsive naming issue will appear more often as we make components that need to adapt well to mobile.

Reply:
@vdepizzol 

> this layout granularity will be needed at some point.

I can absolutely see the value/need to be able to be adjusted depending on the content, and I think it's valuable for us to incorporate. As long as we can find some way to match the visual order to the reading order for screen readers, we're good to go.

Reply:
> On panePosition and paneResponsivePosition

Okay, this makes more sense now! I was assuming you would set the prop once and let the CSS do its thing, but I think what you're saying is you may want to intentionally change the behavior for mobile. I'm calling it mobile! _sigh_

`<Layout panePosition="start" responsivePanePosition="end" />`

Definitely in favor of leaving in the granular props with defaults.

I keep going in circles on the naming thing. The good news is renaming a prop later isn't the end of the world, so I wouldn't really want to hold back this component over it. We're setting props to impact specific window sizes, which means we could eventually add things for large windows- not just narrow (which I think it what you're saying in your last point @vdepizzol!) We don't need to future proof.. but we want to make this flexible for iteration.

I think in this context we're using the word `responsive` to mean `reacts to resizing the window` but its one of those terms that means different things to different people.

`responsiveVariant`: props could be generic (as they are now) or specific to the size `reparateRegionsNarrow` `someFuturePropWide` etc.

`narrowVariant`: only props for narrow screens, so we might need `wideVariant` later

Does changing the word `variant` to `behavior` help in any way? Maybe makes it feel more about resizing and flow vs. something static? All this to say, I think I'm leaning towards keeping the props as you have them now instead of specifically saying `narrow` in the prop name, unless you want to add it to the prop option for more specificity and future proofing.

Reply:
@langermank and I talked a little more over this, and with these suggestions I think the modifier classes and prop names are now much easier to parse: 🤗 🤗 

- Use `Narrow` as a suffix for properties that are only applied on `narrow viewports` (that's a new concept for viewports that only fit one column, such as... `mobile` -- I'll open an issue about it)
- Use `responsive` instead of `variant` for indicating [responsive variants](https://primer.style/css/support/breakpoints#responsive-variants) for extra clarity.
- Simplify prop prop names such as `primaryRegion`
- Add `inherit` as default value for all `dividerNarrow` props

A changelog for @jklina:

### CSS classes
| Old | New |
|--------|--------|
`PageLayout--variant-stackRegions` | `PageLayout--responsive-stackRegions`
`PageLayout--variant-separateRegions` | `PageLayout--responsive-separateRegions`
`PageLayout--variant-stackRegions-panePos-*` |`PageLayout--responsive-panePos-*`
`PageLayout--variant-separateRegions-primary-*` | `PageLayout--responsive-primary-*`
`PageLayout-region--hasDivider-*` | `PageLayout-region--dividerNarrow-*`

### Prop names
| Old | New |
|--------|--------|
`responsivePrimaryRegion` | `primaryRegion`
`paneResponsivePosition` | `panePositionNarrow`
`paneResponsiveDivider` | `paneDividerNarrow`
`headerResponsiveDivider` | `headerDividerNarrow`
`footerResponsiveDivider` | `footerDividerNarrow`

### Args
- `*DividerNarrow` props have new `inherit` value by default